backend/app/api/src/services/UitpasService.ts

```diff
@@ -0,0 +1,395 @@
+                    // event.type ==='deteled' -> not needed as foreign key will delete the order
+                    if (event.type === 'created' && shouldReserveUitpasNumbers(event.model.status)) {
+                        await this.createUitpasNumbers(event.model);
+                        return;
+                    }
+                    if (event.type === 'updated') {
+                        if (event.changedFields.status) {
+                            const statusBefore = event.originalFields.status as OrderStatus;
+                            const statusAfter = event.changedFields.status as OrderStatus;
+                            const shouldReserveAfter = shouldReserveUitpasNumbers(statusAfter);
+                            if (shouldReserveUitpasNumbers(statusBefore) !== shouldReserveAfter) {
+                                if (shouldReserveAfter) {
+                                    await this.createUitpasNumbers(event.model);
+                                    return;
+                                }
+                                await this.deleteUitpasNumbers(event.model);
+                                return;
+                            }
+                        }
+                        if (event.changedFields.data) {
+                            const oldOrder = event.getOldModel() as Order;
+                            if (areUitpasNumbersChanged(oldOrder, event.model)) {
+                                await this.updateUitpasNumbers(event.model);
+                            }
+                        }
+                    }
+                }
+            }
+            catch (e) {
+                console.error('Failed to update UiTPAS-numbers after order update', e);
+            }
+        });
+    }
+
+    static async getSocialTariffForUitpasNumber(uitpasNumber: string, uitpasEventId: string) {
+        // https://docs.publiq.be/docs/uitpas/uitpas-api/reference/operations/list-tariffs
+    }
+
+    static async getSocialTariffForEvent(basePrice: number, uitpasEventId: string) {
+        // https://docs.publiq.be/docs/uitpas/uitpas-api/reference/operations/get-a-tariff-static
+    }
+
+    static async registerTicketSales() {
+        // https://docs.publiq.be/docs/uitpas/uitpas-api/reference/operations/create-a-ticket-sale
+    }
+
+    static async cancelTicketSale() {
+        // https://docs.publiq.be/docs/uitpas/uitpas-api/reference/operations/delete-a-ticket-sale
+    }
+
+    static async getTicketSales() {
+        // https://docs.publiq.be/docs/uitpas/uitpas-api/reference/operations/list-ticket-sales
+    }
+
+    static async searchUitpasEvents() {
+        // input = client credentials of organization & uitpasOrganizerId
+        // https://docs.publiq.be/docs/uitpas/events/searching#searching-for-uitpas-events-of-one-specific-organizer
+    }
+
+    static async searchUitpasOrganizers(name: string, version: number): Promise<UitpasOrganizersResponse> {
```

Comment:
Ik begrijp het nut van de version maar er zijn enkele problemen:
- De UitpasOrganizersResponse structure wordt ook teruggegeven als antwoord op de endpoint - dus de versie moet sowieso in lijn liggen met de Stamhoofd versie.
- Ofwel maak je twee structures met elk een eigen versie nummering (waarvan de structure die je vanaf de API leest enkel in de backend staat en niet geexposed wordt), ofwel decode je altijd met version = Version (constante geimporteerd van de structures package)